src/libcollections/vec.rs

```diff
@@ -247,7 +250,7 @@ impl<T> Vec<T> {
     #[stable(feature = "rust1", since = "1.0.0")]
     pub unsafe fn from_raw_parts(ptr: *mut T, length: usize,
                                  capacity: usize) -> Vec<T> {
-        Vec { ptr: NonZero::new(ptr), len: length, cap: capacity }
+        Vec { ptr: NonZero::new(ptr).unwrap(), len: length, cap: capacity }
```

Comment:
This function should take `NonZero<*mut T>` - why is it stable? Or at least, if we want to keep the signature, keep the current `NonZero::new` as `NonZero::new_unchecked` - the former can be safe, while the latter isn't.
We should also document that passing a null pointer to this function is UB.